packages/journey-manager/src/index.ts

```diff
@@ -211,17 +211,29 @@ export const run = (props: RunProps): RunOutput => {
 
   const triggeredSteps = getTriggeredSteps(props.config.conversationId);
 
-  // If all triggers have a URL constraint set and none match the current URL, fire a digression calback
+  /**
+   * Digression detection
+   * - if all triggers have a URL constraint set and none match the current URL, fire a digression callback
+   * - notable exception: if the step is used as escalation or end as well, missing URL constraint is ok.
+   */
 
   const urlConditions: UrlCondition[] = filterMap(
     Object.values(props.triggers),
     (trigger) => trigger.urlCondition,
   );
 
-  if (
-    Object.keys(props.triggers).length === urlConditions.length &&
-    !urlConditions.some(matchesUrlCondition)
-  ) {
+  // If there are any steps for which there is no URL condition while also not being used for escalation or end,
+  // the package assumes that a digression cannot be reliably detected.
+  const isDigressionDetectable = !Object.entries(props.triggers).some(
+    ([stepId, trigger]) => {
+      return (
+        trigger.urlCondition == null &&
+        !(stepId === props.ui?.escalationStep || stepId === props.ui?.endStep)
+      );
+    },
+  );
+
+  if (isDigressionDetectable && !urlConditions.some(matchesUrlCondition)) {
```

Comment:
💭 minor nit: you can avoid all of the computation by checking if `onDigression` is set before checking for a digression.

Reply:
In the future there will be automatic digression handling (later today actually), I think I would leave this as it is.